app/src/main/java/org/mozilla/fenix/settings/address/ext/Address.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.address.ext
+
+import mozilla.components.concept.storage.Address
+
+/**
+ * Generate a label from a given [Address]. A well-formed [Address] will have
+ * a full name in the label, otherwise it will find the highest priority [Address] property
+ * to use, as based on the desktop code found in: FormAutofillUtils::getAddressLabel
+ */
+fun Address.toLabel(): String = getName().ifEmpty { iterateFields("") }
+
+/**
+ * Generate a description from a given [Address]. A well-formed address will have a street address
+ * as a description. An address with missing data will use the next available property after [label]
+ * based on the priorities found in the desktop code in: FormAutofillUtils::getAddressLabel
+ */
+fun Address.toDescription(label: String): String = iterateFields(label)
+
+private fun Address.getName(): String {
+    val middleName = if (additionalName.isEmpty()) "" else "$additionalName "
+    return if (givenName.isNotEmpty()) "$givenName $middleName$familyName" else familyName
```

Comment:
Could we simplify this to something like `[givenName, middleName, familyName].join(" ")`

Reply:
It would be theoretically possible with an implementation like that to end up with a label that was `middleName familyName`. That's what I was trying to guard against with this slightly awkward logic, but if we aren't concerned about that case then I'm happy to update it.

Reply:
Can you find out what the desktop code is doing for the name label? Your current link only links to the address label generation. We want to emulate as much of the desktop code as possible to account for all the various edge cases when it comes to addresses since we aren't really the expert here.

Reply:
I believe the `getAddressLabel` function includes the `name` parameter of the `address` object they're using on desktop, if that's what you're asking. If you're looking for source on where they split and combine the `name` property into `given`, `additional`, and `family`, I've been struggling to track it down. Haven't used JS for about 4 years so I'm shaking some of the rust off 😅 

I can keep looking for it some tomorrow if needed

Reply:
I finally found the name combination and splitting [here](https://searchfox.org/mozilla-central/source/toolkit/components/formautofill/FormAutofillNameUtils.jsm#400). It looks like the combination is the a simple `join` on non-empty parts, as is reflected in this PR. Splitting the name seems like it will be important for saving addresses though.